test-case-reporting/app.ts

```diff
@@ -32,10 +32,21 @@ function renderTestRunList(testRuns: Array<TestRun>): string {
   return `Rendering ${testRuns.length} test runs!`;
 }
 
+function handleError(error: Error, res: express.Response): void {
+  console.warn(error);
+  if (error instanceof TypeError) {
```

Comment:
nit: if you're using this for a catch-all catch block, it's possible for `TypeError`s to bubble up unrelated to pagination. The best thing to do here is probably to extend `Error` with our own `PaginationError` and check for that. I don't really think it's worth that effort though tbh.

Reply:
I think characterizing bad input as a `TypeError` makes sense -- this isn't only for pagination errors, it also catches bad `buildNumber`s and bad `testCaseId`s. I don't know if I'm being a bit naïve in assuming nothing else will cause `TypeError`s, though.

Reply:
It is certainly possible some other part of the code could throw a typeerror, though using TS obviously makes that much less likely. Like I said, I don't think it's worth the effort to change, just wanted to acknowledge the risk